app/controllers/merchant/invoices_controller.rb

```diff
@@ -1,5 +1,11 @@
 class Merchant::InvoicesController < ApplicationController
 
   def index
+    @merchant = Merchant.find(params[:merchant_id])
+    @invoices = @merchant.invoices.all.uniq
```

Comment:
You may want to turn this into a module method.

Reply:
Agreed. Definitely model logic